### PR TITLE
Fixes #581 Underscores in answer Id submission failure.

### LINKED ISSUE #581: Underscores in answer_ids cause submitter to fail
### Expected behaviour

A schema that uses underscores within answer_ids in the survey json is able to have them submitted as per standard behaviour or a schema that uses underscores fails gracefully (or is unable to be selected?) upon validation.

### Actual behaviour

Submission fails with 'app.schema.exceptions.QuestionnaireException: Unknown id ' message.

### Steps to reproduce the behaviour

Create a survey using an answer_id with an underscore. Attempt to follow through the survey and submit.

### Technical information

This is failing because we use underscores to delimit between an answer_id and answer_instance when calling .map(). We should perhaps update the converter to use the answer_store, rather than the flattened list of answers. https://github.com/ONSdigital/eq-survey-runner/blob/master/app/submitter/converter.py#L87

### PR DESCRIPTION
**Pull request by samiwel**
### What is the context of this PR?
Fixes #581 .

Submitter was doing too simplistic a match to determine the answer instance id.

### How to review 
Submission of answers containing _ should succeed.

### CODE REVIEW

File: app/submitter/converter.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 from datetime import datetime, timezone
 
 from app import settings
+from app.questionnaire_state.state_repeating_answer_question import _extract_answer_instance_id
```
Comment (jonnyshaw89) [inline app/submitter/converter.py]: the underscore at the start of _extract_answer_instance_id implies that this is a "private" function?

File: app/submitter/converter.py [hunk 2]
```diff
@@ -84,8 +86,8 @@ def prepare_answers(metadata_store, questionnaire, answers):
         data = {}
 
         for key in answers.keys():
-            key_without_index = key.rsplit('_', 1)[0]
-            item = questionnaire.get_item_by_id(key_without_index)
+            answer_id, answer_index = _extract_answer_instance_id(key)
```
Comment (jonnyshaw89) [inline app/submitter/converter.py]: answer_index is unused, Codacy will complain?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/submitter/converter.py (hunk 2 above)
Comment (samiwel) [reply app/submitter/converter.py]: It did, resolved.

File: app/submitter/converter.py (hunk 1 above)
Comment (iwootten) [reply app/submitter/converter.py]: I like how the review of the thing to select the right underscore is to remove the underscore from the method name :)

Comment (samiwel) [inline app/submitter/converter.py]: Renamed 👍

**Review by jonnyshaw89 — APPROVED** [review]

**Review by iwootten — COMMENTED** [review]
LGTM 👍

**Review by iwootten — APPROVED** [review]

**Pull request merged**